config/routes.rb

```diff
@@ -1,6 +1,11 @@
 Rails.application.routes.draw do
   get '/', to: 'users#index'
   get '/users/:id', to: 'users#show'
+  get '/posts/new', to: 'posts#new', as: 'new_post'
   get '/users/:user_id/posts', to: 'posts#index'
+  get '/users/:user_id/posts/:id/new_comment', to: 'comments#new', as: 'new_comment'
   get '/users/:user_id/posts/:id', to: 'posts#show'
+  post '/users/1/posts', to: 'posts#create'
+  post '/users/:user_id/posts/:id', to: 'comments#create'
+  post '/users/:user_id/posts/:id/like', to: 'posts#like'
```

Comment:
- [ ] [optional] It would be nice if you used resources for your routing, it makes the routes neat and easier to create and also easier to scale your application, here is an[ article](https://launchschool.com/books/demystifying_rails/read/routes_and_resources) to understand more about resources.